shotover-proxy/tests/kafka_int_tests/mod.rs

```diff
@@ -388,32 +387,127 @@ async fn cluster_2_racks_multi_shotover(#[case] driver: KafkaDriver) {
-    test_cases::cluster_test_suite(&connection_builder).await;
+    produce_consume_partitions3(&connection_builder, "partitions3_rf3", 1, 500).await;
+
+    // Kill one shotover node
+    tokio::time::timeout(
+        Duration::from_secs(10),
+        shotovers.remove(0).shutdown_and_then_consume_events(&[]),
+    )
+    .await
+    .expect("Shotover did not shutdown within 10s");
+
+    // Wait for the other shotover nodes to detect the down node
+    tokio::time::sleep(Duration::from_secs(5)).await;
+
+    // Send more produce and consume requests
+    produce_consume_partitions3(&connection_builder, "partitions3_rf3", 1, 500).await;
+
+    let mut expected_events = multi_shotover_events();
+    // Other shotover nodes should detect the killed node at least once
+    expected_events.push(
+        EventMatcher::new()
+            .with_level(Level::Warn)
+            .with_target("shotover::transforms::kafka::sink_cluster::shotover_node")
+            .with_message(r#"Shotover peer 127.0.0.1:9191 is down"#)
+            .with_count(Count::GreaterThanOrEqual(1)),
+    );
 
     for shotover in shotovers {
         tokio::time::timeout(
             Duration::from_secs(10),
-            shotover.shutdown_and_then_consume_events(&multi_shotover_events()),
+            shotover.shutdown_and_then_consume_events(&expected_events),
         )
         .await
         .expect("Shotover did not shutdown within 10s");
     }
 }
+
 #[rstest]
 #[cfg_attr(feature = "kafka-cpp-driver-tests", case::cpp(KafkaDriver::Cpp))]
 #[case::java(KafkaDriver::Java)]
 #[tokio::test(flavor = "multi_thread")] // multi_thread is needed since java driver will block when consuming, causing shotover logs to not appear
-async fn cluster_2_racks_multi_shotover_with_one_shotover_down(#[case] _driver: KafkaDriver) {
+async fn cluster_3_racks_multi_shotover_with_2_shotover_down(#[case] driver: KafkaDriver) {
     let _docker_compose =
-        docker_compose("tests/test-configs/kafka/cluster-2-racks/docker-compose.yaml");
+        docker_compose("tests/test-configs/kafka/cluster-3-racks/docker-compose.yaml");
+    let mut shotovers = vec![];
+    for i in 1..4 {
+        shotovers.push(
+            shotover_process(&format!(
+                "tests/test-configs/kafka/cluster-3-racks/topology-rack{i}.yaml"
+            ))
+            .with_config(&format!(
+                "tests/test-configs/shotover-config/config{i}.yaml"
+            ))
+            .with_log_name(&format!("shotover{i}"))
+            .start()
+            .await,
+        );
+    }
 
-    // One shotover instance per rack
+    // Wait for check_shotover_peers to start
+    tokio::time::sleep(Duration::from_secs(15)).await;
+
+    // Send some produce and consume requests
+    let connection_builder = KafkaConnectionBuilder::new(driver, "localhost:9193");
+    produce_consume_partitions3(&connection_builder, "partitions3_rf3", 1, 500).await;
```

Comment:
Its worth noting that I dont think this is actually running against a topic with rf3 and 3 partitions.
Since `admin_setup` hasnt been called, its relying on kafka's auto topic creation which I believe creates a topic with RF1 and 1 partition.

But more importantly I think we need to change these tests to hold a connection open while the shotover node is killed.
See for example `produce_consume_partitions1_kafka_node_goes_down` which:
1. creates a producer and consumer
2. kills a kafka node
3. continues to use the producer and consumer.

Judging from produce_consume_partitions1_kafka_node_goes_down it looks like we'll have to skip this test on the cpp driver since it doesnt handle this case well.

It might be worth also testing creating new connections after the shotover node is killed, as you are doing currently, maybe we could run the standard test suite instead, that way we properly run admin_setup.

Reply:
I'm thinking the best way to implement it would be to define a test_case function similar to produce_consume_partitions1_kafka_node_goes_down which takes a `Vec<BinProcess>` of shotover nodes to kill.